src/tensorwaves/optimizer/minuit.py

```diff
@@ -119,11 +119,12 @@ def wrapped_gradient(pars: list) -> Iterable[float]:
             )
         )
 
-        return {
-            "minimum_valid": minuit.valid,
-            "parameter_values": parameter_values,
-            "parameter_errors": parameter_errors,
-            "log_likelihood": minuit.fmin.fval,
-            "function_calls": minuit.fmin.nfcn,
-            "execution_time": end_time - start_time,
-        }
+        return FitResult(
+            minimum_valid=minuit.valid,
+            execution_time=end_time - start_time,
+            function_calls=minuit.fmin.nfcn,
+            estimator_value=minuit.fmin.fval,
+            parameter_values=parameter_values,
+            parameter_errors=parameter_errors,
+            specifics=minuit,
```

Comment:
I don't like that the whole minuit object is attached to the fit result, because:

1. Its an object that does not represent only a fit result, but the whole minimizer
2. It contains duplicate information which is already available in the other attributes.

(see also comment on FitResult)

Reply:
Yes, that's true. As for 1., that's just the unfortunate design `iminuit` (the optimizer serves as a fit result container as well).